platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -849,6 +849,8 @@ export default class HangingProtocolService extends PubSubService {
 
       throw new Error(error);
     }
+
+    this._commandsManager.run(this.protocol.callbacks?.onProtocolEnter);
```

Comment:
The one below for onProtocolExit has an extra null check, so consider making them consistent.

`this._commandsManager.run(this?.protocol.callbacks?.onProtocolEnter);`